pkg/ingestor/parser/spdx/parse_spdx.go

```diff
@@ -123,9 +123,8 @@ func (s *spdxParser) getPackages() error {
 
 		if slices.Contains(topLevelSpdxIds, string(pac.PackageSPDXIdentifier)) {
```

Comment:
This looks good but does this check for the `getTopLevelPackageSpdxIds` need to be changed? @lumjjb?

https://github.com/guacsec/guac/blob/07353d5563c6690415225c8b96ee388e0e98317f/pkg/ingestor/parser/spdx/parse_spdx.go#L89-L93

Reply:
AFAIU, this looks good as it's properly "contributing" to the `HasSBOM` predicate's creation during SBOM ingestion.
But obviously @lumjjb knows better.

At the same time, the issue and the fix here are more towards let the `getPackageElement` func to work properly since it's the one in charge of finding the `{found,related}PackNodes` values and this feature is based only searching within the `packagePackages` map.

I hope this helps.

Reply:
Just did some testing to understand. Looks good.